### DISCUSSION TITLE
pyinstaller package well but could not open exe

### DISCUSSION THREAD

**Discussion by wanghaisheng** [fastapi/fastapi#8482] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
https://github.com/wanghaisheng/link-detec-shopconan/blob/main/main.py
```

### Description

Traceback (most recent call last):
  File "app.py", line 3, in <module>
  File "D:\Program Files\anaconda3\lib\site-packages\fastapi\__init__.py", line 7, in <module>
    from .applications import FastAPI as FastAPI
  File "D:\Program Files\anaconda3\lib\site-packages\fastapi\applications.py", line 4, in <module>
    from fastapi import routing
  File "D:\Program Files\anaconda3\lib\site-packages\fastapi\routing.py", line 21, in <module>
    from fastapi import params
  File "D:\Program Files\anaconda3\lib\site-packages\fastapi\params.py", line 4, in <module>
    from pydantic.fields import FieldInfo, Undefined
  File "pydantic\__init__.py", line 2, in init pydantic.__init__
ImportError: cannot import name dataclasses

### Operating System

Windows

### FastAPI Version

0.76.0

### Python Version

3.8.5

**Comment by acidjunk** [comment]
Does pyinstaller still use tree shaking to get a small binary?

I can imagine that the Cython part in Pydantic can cause problems:
https://pydantic-docs.helpmanual.io/install/

So pyinstaller might not package the pydantic stuff correctly (just my 2 cent)

**Comment by wanghaisheng** [comment]
that is helpful
i will try another machine /my mac +docker later and update more details
@acidjunk
